### ISSUE TITLE
Upgrade Jackson  to > 2.9.10.7

### ISSUE THREAD

**Issue by yeikel** [joelittlejohn/jsonschema2pojo#1229]
Currently we are using [Jackson 2.9.10.7 ](https://github.com/joelittlejohn/jsonschema2pojo/blob/master/pom.xml#L56), but this version has many vulnerabilities[1-2]

1. https://github.com/advisories/GHSA-5949-rw7g-wx7w
2. https://nvd.nist.gov/vuln/detail/CVE-2021-20190

Ideally we should upgrade to > 2.9.10.7

**Comment by joelittlejohn** [comment]
Your link number 1 seems to relate to an issue that is fixed in 2.9.10.7.

Since we don't actually dictate which Jackson version is used when deploying this generated code, these widget vulnerabilities are largely irrelevant to us. End users are free to use whatever Jackson version they like (and the version used by this plugin cannot help or guarantee that the version used in your project is recent and secure).

However I'm all for keeping things up to date. I think the way we have a Maven variable here is maybe confusing dependabot. Feel free to raise a PR that updates that version and I will merge.

**Comment by joelittlejohn** [comment]
Upgraded. Thanks for raising this 👍

**Comment by yeikel** [comment]
I am not sure if I am doing anything wrong, but this is what I have : 


```
  <plugin>
                <groupId>org.jsonschema2pojo</groupId>
                <artifactId>jsonschema2pojo-maven-plugin</artifactId>
                <version>1.1.0</version>
                <configuration>
                    <sourceDirectory>${basedir}/src/main/resources/models</sourceDirectory>
                    <targetPackage>com.models</targetPackage>
                    <sourceType>json</sourceType>
                    <annotationStyle>gson</annotationStyle>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```
And when I try to build the project I see that it is trying to pull `com/fasterxml/jackson/core/jackson-databind/2.9.10.7/jackson-databind-2.9.10.7.jar `

**Comment by yeikel** [comment]
`[ERROR] Failed to execute goal org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.1.0:generate (default) on projecttools: Execution default of goal org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.1.0:generate failed: Plugin org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.1.0 or one of its dependencies could not be resolved: Could not transfer artifact com.fasterxml.jackson.core:jackson-databind:jar:2.9.10.7 from proxy/content/groups/prod/com/fasterxml/jackson/core/jackson-databind/2.9.10.7/jackson-databind-2.9.10.7.jar 403 Requested item is quarantined -> [Help 1]
`

So it seems that you are still depending on that deprecated version

**Comment by yeikel** [comment]
I also cloned the repository locally and tried to build the `master` branch, and it failed as well. It seems that one of your transitive dependencies is using that version

**Comment by yeikel** [comment]
```
➜  jsonschema2pojo git:(master) mvn clean install
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] jsonschema2pojo                                                    [pom]
[INFO] jsonschema2pojo-core                                               [jar]
[INFO] jsonschema2pojo-ant                                                [jar]
[INFO] jsonschema2pojo-cli                                                [jar]
[INFO] jsonschema2pojo-gradle-plugin                                      [jar]
[INFO] jsonschema2pojo-maven-plugin                              [maven-plugin]
[INFO] jsonschema2pojo-integration-tests                                  [jar]
[INFO] 
[INFO] ----------------< org.jsonschema2pojo:jsonschema2pojo >-----------------
[INFO] Building jsonschema2pojo 1.1.1-SNAPSHOT                            [1/7]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ jsonschema2pojo ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.2:enforce (enforce-maven) @ jsonschema2pojo ---
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ jsonschema2pojo ---
[INFO] Updating license headers...
Downloading from prod: java-proxy/content/groups/prod/com/fasterxml/jackson/core/jackson-databind/2.9.10.8/jackson-databind-2.9.10.8.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for jsonschema2pojo 1.1.1-SNAPSHOT:
[INFO] 
[INFO] jsonschema2pojo .................................... FAILURE [  4.544 s]
[INFO] jsonschema2pojo-core ............................... SKIPPED
[INFO] jsonschema2pojo-ant ................................ SKIPPED
[INFO] jsonschema2pojo-cli ................................ SKIPPED
[INFO] jsonschema2pojo-gradle-plugin ...................... SKIPPED
[INFO] jsonschema2pojo-maven-plugin ....................... SKIPPED
[INFO] jsonschema2pojo-integration-tests .................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.678 s
[INFO] Finished at: 2021-03-23T17:49:10-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project jsonschema2pojo-core: Could not resolve dependencies for project org.jsonschema2pojo:jsonschema2pojo-core:jar:1.1.1-SNAPSHOT: Could not transfer artifact com.fasterxml.jackson.core:jackson-databind:jar:2.9.10.8 from/to prod (java-proxy/content/groups/prod/): Authorization failed for java-proxy/content/groups/prod/com/fasterxml/jackson/core/jackson-databind/2.9.10.8/jackson-databind-2.9.10.8.jar 403 Requested item is quarantined -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :jsonschema2pojo-core
```

**Comment by joelittlejohn** [comment]
> This is what I have :
> ...

Any build you run that depends on jsonschema2pojo v1.1.0 will still use jackson-databind 2.9.10.7. When 1.1.1 is released it will depend on the new version 2.9.10.8.

> I also cloned the repository locally and tried to build the master branch
> ...

It looks like your proxy is blocking jackson-databind 2.9.10.8 from being downloaded.
